src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -145,16 +145,24 @@ public bool TryLoadConfig(
         {
             Console.WriteLine($"Loading config file from {path}.");
             string json = _fileSystem.File.ReadAllText(path);
-            bool configParsed = TryParseConfig(json, out RuntimeConfig, connectionString: _connectionString, replaceEnvVar: replaceEnvVar);
-            TrySetupConfigFileWatcher();
-
-            if (RuntimeConfig is not null && !string.IsNullOrEmpty(defaultDataSourceName))
+            if (TryParseConfig(json, out RuntimeConfig, connectionString: _connectionString, replaceEnvVar: replaceEnvVar))
             {
-                RuntimeConfig.DefaultDataSourceName = defaultDataSourceName;
+                if(TrySetupConfigFileWatcher())
+                {
+                    logger?.LogInformation("Monitoring config: {ConfigFilePath} for hot-reloading.", ConfigFilePath);
+                }
+
+                if (!string.IsNullOrEmpty(defaultDataSourceName))
+                {
+                    RuntimeConfig.DefaultDataSourceName = defaultDataSourceName;
+                }
+
+                config = RuntimeConfig;
+                return true;
```

Comment:
It makes sense that `TrySetupConfigFileWatcher()` is best effort. Within that function, are there any logging mechanisms to emit information explaining why it may have failed in a failure case?

Reply:
we pass along the exception message if the instantiation of the file watcher fails